### PR TITLE
feat: [PM4-73] admin parkplatzliste UI

### PR DESCRIPTION
**Pull request by SandroGuerotto**
Created parkinglot admin page.
Changed navbar to support admin and user role.

### CODE REVIEW

File: frontend/pages/login.tsx [hunk 1]
```diff
@@ -71,7 +80,7 @@ export default function Login() {
           const data = await res.json();
           await mutateUser({
             isLoggedIn: true,
-            roles: data.roles,
+            role: data.roles[0],
```
Comment (DaRabus) [inline frontend/pages/login.tsx:83]: Besch secher das mer nor das Attribute 0 wöndt notze vode Rollene? ech glaub mer bechömit doch en Array full of Enums, oder?

File: frontend/pages/admin/parking-lots/index.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    if (!user) return;
+    setParkingLots({ error: null, loading: true, result: [] });
+    fetchParkingLots(false, user)
+      .then((result) => {
+        if (result) {
+          setParkingLots({ error: null, loading: false, result: result });
+        }
+      })
+      .catch();
+  }, []);
```
Comment (DaRabus) [inline frontend/pages/admin/parking-lots/index.tsx]:
```suggestion
  useEffect(() => {
    if (user) {
    setParkingLots({ error: null, loading: true, result: [] });
    fetchParkingLots(false, user)
      .then((result) => {
        if (result) {
          setParkingLots({ error: null, loading: false, result: result });
        }
      })
      .catch();
      }
  }, []);
```

File: frontend/pages/admin/parking-lots/index.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          changeParkingLotState={changeParkingLotState}
+        />
+      ];
+    });
+
+  return (
+    <Grid padding={2}>
+      <Grid item xs={12}>
+        <ParkingLotsFilter
+          updateFilter={(newFilter) => setFilter(() => newFilter)}
```
Comment (DaRabus) [inline frontend/pages/admin/parking-lots/index.tsx]:
```suggestion
          updateFilter={setFilter}
```

Should work like this if the Actionstate is set

File: frontend/pages/admin/parking-lots/index.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      </Grid>
+
+      <Grid item xs={12}>
+        <>
+          <Loading loading={parkingLots.loading} />
+
+          {parkingLots.result.length > 0 ? (
+            <ParkingLotsTable
+              parkingLots={filteredReservations}
+            ></ParkingLotsTable>
```
Comment (DaRabus) [inline frontend/pages/admin/parking-lots/index.tsx]: JSX Elements can be shortend when no children are provided

```suggestion
            <ParkingLotsTable
              parkingLots={filteredReservations}
            />
```

File: frontend/pages/login.tsx [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
+import { useRouter } from 'next/router';
 
 export default function Login() {
+  const router = useRouter();
   // here we just check if user is already logged in and redirect to profile
-  const { mutateUser, user } = useUser({
-    redirectTo: '/search',
-    redirectIfFound: true
-  });
+  const { mutateUser, user } = useUser({});
```
Comment (DaRabus) [inline frontend/pages/login.tsx]: Dont need the empty object, should actually throw you an eslint error?

```suggestion
  const { mutateUser, user } = useUser();
```

File: frontend/pages/search/index.tsx [hunk 6]
```diff
@@ -47,18 +48,17 @@ const SearchPage = () => {
     setShowDetails(true);
   };
 
-  const makeOnSearch = async(searchParameters: SearchParameters) => {
+  const makeOnSearch = (searchParameters: SearchParameters) => {
     if (!user) return;
```
Comment (DaRabus) [inline frontend/pages/search/index.tsx:52]: Rather check if user is existing instead of returning null

File: frontend/pages/search/index.tsx [hunk 7]
```diff
@@ -47,18 +48,17 @@ const SearchPage = () => {
...
-    await fetchParkingSpots(searchParameters, true, user)
+    fetchParkingSpots(searchParameters, true, user)
       .then((result) => {
         setSearchResult({ error: null, loading: false, result: result });
       })
       .catch((error) => toast.error(error.message));
   };
 
-  const mappedResult: Array<string[]> = searchResult.result.map((item) => {
+  const mappedResult: Array<RowDataType> = searchResult.result.map((item) => {
```
Comment (DaRabus) [inline frontend/pages/search/index.tsx:61]:
```suggestion
  const mappedResult: RowDataType[] = searchResult.result.map((item) => {
```

File: frontend/src/components/navbar/drawer/drawer-admin.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+import { ColorModeContext } from '../../../../context';
+
+function DrawerComponent() {
+  const classes = useStyles();
+  const [openDrawer, setOpenDrawer] = useState(false);
+  const colorMode = useContext(ColorModeContext);
+  const { user } = useUser({
+    redirectTo: '/login',
+    redirectIfFound: false
+  });
```
Comment (DaRabus) [inline frontend/src/components/navbar/drawer/drawer-admin.tsx:24]: Mer hend das scho im Layout de redirect?

Settit das ned öberall dri tue, sost gets mega vell rerenders :D

File: frontend/src/components/navbar/drawer/drawer-admin.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+          </ListItem>
+          <ListItem onClick={() => setOpenDrawer(false)}>
+            <ListItemText>
+              <Link href="/admin/users">Benutzerverwaltung</Link>
+            </ListItemText>
+          </ListItem>
+          <ListItem onClick={() => setOpenDrawer(false)}>
+            <ListItemText>
+              <Link href="/logout">
+                <span className={classes.logout}>
```
Comment (DaRabus) [inline frontend/src/components/navbar/drawer/drawer-admin.tsx:42]: Typography wäri sost de MUI component för schrifte, de hesch au die richtigi FONT

File: frontend/src/components/parking-detail-modal/parking-detail-modal.tsx [hunk 10]
```diff
@@ -85,13 +84,34 @@ const ParkingDetailModal = ({
...
-                        <FormControlLabel label="Sa" control={<Checkbox checked />} />  
-                        <FormControlLabel label="So" control={<Checkbox checked />} />   
+                        <FormControlLabel
+                          label="Mo"
+                          control={<Checkbox checked />}
+                        />
+                        <FormControlLabel
+                          label="Di"
+                          control={<Checkbox checked />}
+                        />
```
Comment (DaRabus) [inline frontend/src/components/parking-detail-modal/parking-detail-modal.tsx:94]: So much spacing?

Ben ned secher, aber lueg mol ob du Eslint ond Prettier settings dinne hesch, het paar komeschi sache bi dir im PR

File: frontend/src/components/parking-detail-modal/parking-detail-modal.tsx [hunk 11]
```diff
@@ -185,11 +205,12 @@ const ParkingDetailModal = ({
...
-            {parkingLotModel.pictures.map((picture) => (
-              <div>
-                <ImageCustom alt="images" src={picture} />
-              </div>
-            ))}
+            {parkingLotModel.pictures &&
+              parkingLotModel.pictures.map((picture) => (
+                <div>
+                  <ImageCustom alt="images" src={picture} />
+                </div>
```
Comment (DaRabus) [inline frontend/src/components/parking-detail-modal/parking-detail-modal.tsx:212]: Div settsh ned dröber bruche, ond möndt no luege, aktuell gsehts so us als send belder im Backend noni definiert?

Es send glaub kei Links, oder sends base64 belder, weiss es ned. Aber das goht beides ned met dene props, je nachdem bruchsh de proxy oder es attribute "localImage"

File: frontend/src/components/parking-list/parking-list.tsx [hunk 12]
```diff
@@ -1,33 +1,35 @@
...
-    const randomColors : string[] = ["red", "blue", "yellow", "green", "purple"];
-    const randomColor = (index : number) : string => {
-      // TODO : update with random color
-      return randomColors[index];
-    };
+  const randomColors: string[] = ['red', 'blue', 'yellow', 'green', 'purple'];
+  const randomColor = (index: number): string => {
+    // TODO : update with random color
+    return randomColors[index];
+  };
```
Comment (DaRabus) [inline frontend/src/components/parking-list/parking-list.tsx:13]: Esch das ned die Story vode Nadine met de ahzeig vode eigne parkplätz ihde farbene?

Oder för was bruchsh die random farbene?

Ahjo, random Color: https://css-tricks.com/snippets/javascript/random-hex-color/

File: frontend/src/components/parking-lots-list/parking-lot-state-toggle-button.tsx [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    <Typography>gesperrt</Typography>
+  ) : (
+    <Typography>sperren</Typography>
+  );
+const releasedText = (state: ParkingLotState) =>
+  state === ParkingLotState.released ? (
+    <Typography>freigegeben</Typography>
+  ) : (
+    <Typography>freigeben</Typography>
+  );
```
Comment (DaRabus) [inline frontend/src/components/parking-lots-list/parking-lot-state-toggle-button.tsx:51]: You could put the Typography in the parent and just pass the string, saves you some extra wrappers

File: frontend/src/components/parking-lots-list/parking-lots-filter.tsx [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+interface ParkingLotsFilterProps {
+  updateFilter: (data: ParkingLotsFilterData) => void;
+}
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
```
Comment (DaRabus) [inline frontend/src/components/parking-lots-list/parking-lots-filter.tsx:25]: Don't forget what happens on the overflow, for example add "overflow: scroll"

File: frontend/src/components/search-parking-lot/search-parking-lot-table.tsx [hunk 15]
```diff
@@ -1,9 +1,9 @@
-import { logger } from '../../logger';
 import TableComponent from '../table/table-component';
 import React from 'react';
+import { RowDataType } from '../table/table-row';
 
 interface SearchParkingLotTableProps {
-  parkingLots: Array<string[]>;
+  parkingLots: Array<RowDataType>;
```
Comment (DaRabus) [inline frontend/src/components/search-parking-lot/search-parking-lot-table.tsx]:
```suggestion
  parkingLots: RowDataType[];
```

File: frontend/src/components/table/table-component.tsx [hunk 16]
```diff
@@ -8,8 +8,8 @@ import TableHeader from './table-header';
 
 interface TableComponentProps {
   headerNames: string[];
-  onRowClick: (row: any) => void;
-  data: RowDataType[];
+  onRowClick?: (row: unknown) => void;
```
Comment (DaRabus) [inline frontend/src/components/table/table-component.tsx:11]: i assume it's a mix between string or boolean?

could also use then 
```suggestion
  onRowClick?: (row: string | boolean) => void;
```

File: frontend/src/components/table/table-component.tsx [hunk 17]
```diff
@@ -8,8 +8,8 @@ import TableHeader from './table-header';
...
+  data: Array<RowDataType>;
```
Comment (DaRabus) [inline frontend/src/components/table/table-component.tsx]:
```suggestion
  data: RowDataType[];
```

File: frontend/src/components/table/table-row.tsx [hunk 18]
```diff
@@ -7,13 +7,18 @@ export type RowDataType = Array<string | JSX.Element>;
 
 interface CustomTableRowProps {
   rowKey: number;
-  onRowClick: (row: any) => void;
+  onRowClick?: (row: unknown) => void;
```
Comment (DaRabus) [inline frontend/src/components/table/table-row.tsx:10]: Same here
```suggestion
  onRowClick?: (row: string | boolean) => void;
```

File: frontend/src/mock-data/search-dummy.tsx [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+      name: 'Blümchen',
+      role: UserRole.user
+    },
+    pictures: [
+      'https://placehold.co/600x600.png',
+      'https://placehold.co/600x600.png'
+    ],
+    price: 1200,
+    state: ParkingLotState.locked,
+    tags: ['nice view', 'nice view', 'nice view', 'nice view']
```
Comment (DaRabus) [inline frontend/src/mock-data/search-dummy.tsx:51]: tbh, i expect that the tags are gonna be ENUM, but didn't hear anything yet from the backend team

**Review by DaRabus — APPROVED** [review]
Just suggestions, since it's just a school project, as long as it works, i don't mind too much D:

File: frontend/pages/login.tsx (hunk 1 above)
Comment (SandroGuerotto) [reply frontend/pages/login.tsx:83]: ja. wir hend im refinement bestimmt, dass 1 user nur 1 rolle ha chan. daher hanis ahpasst

File: frontend/src/mock-data/search-dummy.tsx (hunk 19 above)
Comment (SandroGuerotto) [reply frontend/src/mock-data/search-dummy.tsx:51]: Would be best to have enum. but for this project it does not matter...

**Pull request merged**